src/pages/signin/SignInPage.js

```diff
@@ -95,42 +110,37 @@ class SignInPage extends Component {
+        const showValidateCodeForm =
+            !showWelcomeForm && (this.props.credentials.login || this.props.credentials.validateCode) && !showUnlinkLoginForm && Permissions.canUsePasswordlessLogins(this.props.betas);
 
         // Show the resend validation link form if
         // - A login has been entered
         // - AND is not validated or password is forgotten
         // - AND the login isn't an unvalidated secondary login
         // - AND user is not on 'passwordless' beta
         const showResendValidationForm =
+            !showWelcomeForm &&
             Boolean(this.props.credentials.login) &&
             (!this.props.account.validated || this.props.account.forgotPassword) &&
             !showUnlinkLoginForm &&
             !Permissions.canUsePasswordlessLogins(this.props.betas);
 
         let welcomeHeader = '';
         let welcomeText = '';
+
+        const userLogin = Str.removeSMSDomain(lodashGet(this.props, 'credentials.login', ''));
+        const userLoginToDisplay = Str.isSMSLogin(userLogin) ? this.props.formatPhoneNumber(userLogin).replace(/ /g, '\u00A0') : userLogin;
+
         if (showValidateCodeForm) {
             if (this.props.account.requiresTwoFactorAuth) {
                 // We will only know this after a user signs in successfully, without their 2FA code
                 welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
                 welcomeText = this.props.translate('validateCodeForm.enterAuthenticatorCode');
-            } else {
-                const userLogin = Str.removeSMSDomain(lodashGet(this.props, 'credentials.login', ''));
-
-                // replacing spaces with "hard spaces" to prevent breaking the number
-                const userLoginToDisplay = Str.isSMSLogin(userLogin) ? this.props.formatPhoneNumber(userLogin).replace(/ /g, '\u00A0') : userLogin;
-                if (this.props.account.validated) {
-                    welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
-                    welcomeText = this.props.isSmallScreenWidth
-                        ? `${this.props.translate('welcomeText.welcomeBack')} ${this.props.translate('welcomeText.welcomeEnterMagicCode', {login: userLoginToDisplay})}`
-                        : this.props.translate('welcomeText.welcomeEnterMagicCode', {login: userLoginToDisplay});
-                } else {
-                    welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcome');
-                    welcomeText = this.props.isSmallScreenWidth
-                        ? `${this.props.translate('welcomeText.welcome')} ${this.props.translate('welcomeText.newFaceEnterMagicCode', {login: userLoginToDisplay})}`
-                        : this.props.translate('welcomeText.newFaceEnterMagicCode', {login: userLoginToDisplay});
-                }
+            } else if (this.props.account.validated) {
```

Comment:
This is starting to feel a bit redundant to me. Is there ever a case where we would have:

- `showValidateCodeForm` - truthy
- `!this.props.account.requiresTwoFactorAuth` 
- `!this.props.account.validated`

?

Said another way, if we are in the `showValidateCodeForm` block and `this.props.account.validated` is false... will we just get no welcome header/text?

Reply:
good catch! changing `else if` to `else`